functions/airtableTriager.js

```diff
@@ -29,54 +120,124 @@ async function onNewIntake(id, fields) {
     })
 
+    // TODO : we should retry as we might be getting throttled
     if (!res["ok"]) {
-        console.error(`Encountered an error posting ticket to #${neighborhoodChanName}: ${id}`)
+        console.error(`Encountered an error posting ticket ${id} to #${neighborhoodChanName}: ${res["error"]}`)
         return
     }
+
+    meta["intakePostChan"] = neighborhoodChanID
+    meta["intakePostTs"] = res["ts"]
 }
 
-// Runs every minute
-exports.poll.airtableIntakeTickets = functions.pubsub.schedule('* * * * *').onRun(async () => {
-    const STATUS_TO_CBS = {
-        "Seeking Volunteer": [onNewIntake],
-        "Assigned / In Progress": [],
-        "Complete": [],
-        "Not Bed-Stuy": [],
+async function onIntakeAssigned(id, fields, meta) {
+    console.log(`onIntakeAssigned(${id})`)
+
+    if (!meta["intakePostChan"] || !meta["intakePostTs"]) {
+        console.error(`This ticket wasn't posted to slack: ${id}`)
+        return
+    }
+
+    let res = await bot.chat.update({
+        channel: meta["intakePostChan"],
+        ts: meta["intakePostTs"],
+        text: await getIntakePostContent(fields),
+    })
+
+    // TODO : we should retry as we might be getting throttled
+    if (!res["ok"]) {
+        console.error(`Encountered an error updating ticket ${id}: ${res["error"]}`)
+        return
     }
 
-    const changedTickets = await getChangedIntakeTickets()
+    meta["intakePostTs"] = res["ts"]
 
-    if (changedTickets.length === 0) {
-        return null
+    // TODO : use the delivery volunteer's id
+    res = await bot.chat.postMessage({
+        channel: "UV5KXJYTC",
+        as_user: true,
+        text: await getDeliveryDMContents(fields),
+        unfurl_media: true,
+    })
+
+    if (!res["ok"]) {
+        console.error(`Encountered an error sending dm for ticket ${id}: ${res["error"]}`)
+        return
     }
 
-    // TODO : it is possible for us to miss a step in the intake ticket state transitions.
-    // A ticket should go from "Seeking Volunteer" -> "Assigned" -> "Complete". Since we
-    // only trigger on the current state, there is a race condition where we could miss
-    // the intermediate state (i.e. assigned)
-    for (const [id, fields] of changedTickets) {
-        console.log(`Processing intake ticket (${fields["Ticket ID"]}): ${id}`)
-
-        const status = fields["Status"]
-        if (!(status in STATUS_TO_CBS)) {
-            console.error(`Ticket has invalid status: ${status}`)
-            continue
-        }
-
-        for (const cb of STATUS_TO_CBS[status]) {
-            // eslint-disable-next-line callback-return
-            await cb(id, fields)
-        }
+    // TODO : post a comment in the thread saying that the delivery has been claimed
+}
+
+async function onIntakeCompleted(id, fields, meta) {
+    console.log(`onIntakeCompleted(${id})`)
+
+    // TODO : we could try recovering from this by calling `onNewIntake` manually
+    if (!meta["intakePostChan"] || !meta["intakePostTs"]) {
```

Comment:
just for clarity for anyone else who needs to work on this can we avoid abbreviations? stuff like channel and timestamp here

Reply:
also idk if you're coming from python but javascript lets you access these directly like `meta.intakePostChannel`